src/components/ShareTrajectoryModal/EmbedPreview.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import CustomModal from "../CustomModal";
+
+import styles from "./style.css";
+
+interface EmbedPreviewProps {
+    embedSnippet: string;
+    height: number;
+    width: number;
+    preview: boolean;
+    setPreview: (value: boolean) => void;
+}
+
+const EmbedPreview: React.FC<EmbedPreviewProps> = ({
+    embedSnippet,
+    height,
+    width,
+    preview,
+    setPreview,
+}) => {
+    document.documentElement.style.setProperty(
+        "--embed-preview-modal-width",
+        `${width + 50}px`
+    );
+    document.documentElement.style.setProperty(
+        "--embed-preview-modal-height",
+        `${height + 112}px`
+    );
+
+    return (
+        <CustomModal
+            closeHandler={() => {
+                setPreview(!preview);
+            }}
+            titleText="Preview"
+            className={styles.embedPreviewModal}
+        >
+            <div dangerouslySetInnerHTML={{ __html: embedSnippet }} />
+        </CustomModal>
```

Comment:
There are a few spots with `dangerouslySetInnerHTML` in our codebase, I want to be careful though.

The html input that generate the snippet is disabled, but is this a security risk? We can use a library like `dompurify` to sanitize the html but I didn't want to add a dependency if not necessary.